CONTRIBUTING.md

```diff
@@ -0,0 +1,75 @@
+# Contribution Guide
+
+## Getting Started
+
+### Prerequisites
+Please make sure you have the following programs installed:
+- [Python 3.9](https://www.python.org/downloads/)
+- [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+- [Virtual Box](https://www.virtualbox.org/wiki/Downloads)
+- [Vagrant](https://www.vagrantup.com/downloads)
+
+### Setting up
+1. Clone your fork.
+2. Create a new branch.
+3. Open any terminal and navigate to the project directory.
+4. Run the `vagrant up` command.
+
+After the dependencies were installed and the VM initialized, you can proceed working on the project code.
+
+## Development Process
+
+### Commit Messages
+- Separate subject from body with a blank line.
+- Limit the subject line to 50 characters.
+- Capitalize the subject line.
+- Do not end subject line with a period.
+- Use the imperative mood in the subject line.
+- Wrap the body at 72 characters.
+- Use the body to explain what and why vs. how.
+
+For more information and examples, please refer to [this guide](https://cbea.ms/git-commit/#seven-rules).
+
+### Issues
+- The title must be clear, and describe what the issue is about. 
+- The description of the issue must include a clear explanation about the reasons for doing the work. Also, it should be as small as possible.
+- Status Labels(Optional):
+  - Status/New - An issue that was just created, in addition unlabeled issues are considered new.
+  - Status/Backlog - A issue that was discussed and accepted for work.
+  - Status/In progress - An issue that is being worked on.
+
+### Pull Requests
+- The Pull Request should not break any of the existing functionality
+- Description: the PR description should be as small as possible, and explain what changes have been made, and why.
+In case it's related to a specific `issue` - it is mandatory to include a link to it.
```

Comment:
To submit a PR, it is mandatory to link it to a relevant issue. This means that if there is no existing issue that relates to your PR, you must create one before you can submit your request.
please make it more implicit in the Pull Requests section

Reply:
I was thinking that "Issue" section is related mainly to report bugs in our code.
That's why I haven't included the part of "This means that if there is no existing issue that relates to your PR, you must create one before you can submit your request.". Because what happens when the contributor creates a new feature? New features would be presented in the issues section?
Thank you for your comment.

Reply:
Its not focused on bugs only , it should also be used to discuss new features or improvements to existing functionality. see more [About issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues)

Reply:
Thank you for your comment. Now it all makes sense. I've updated the file, and a new commit will appear soon.